hazelcast/src/test/java/testsubjects/CustomExceptions.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package testsubjects;
+
+public class CustomExceptions {
```

Comment:
Probably it makes sense to add an exception class with non-matching signature and check that it leads to a `UndefinedErrorCodeException`. Or we could simply keep `ClientProtocolErrorCodesTest#testUndefinedException`. WDYT?

Reply:
`testUndefinedException` does not work anymore. 
When you put it this way, I have just realized that the meaning of `UndefinedErrorCodeException` has changed also. The Javadoc says
```
This exception is thrown when an exception that is coming from the server is not recognized by the protocol.
 * Class name of the original exception is included in the exception
```
which is not correct anymore. 

I will change to 
```
 * This exception is thrown when an exception that is coming from server is not recognized by the protocol and
 * it can not be constructed by the client via reflection.
 * For the client to be able to recreate original exception it should be available on the classpath and
 * it should have one of the following constructors publicly.
 * new Throwable(String message, Throwable cause)
 * new Throwable(Throwable cause)
 * new Throwable(String message)
 * new Throwable()
 * <p>
 * Class name of the original exception is included in the exception.
````

And I can add an exception class with a non-matching signature as you suggested to verify the behavior.